src/sql/arrow_sql_gen/postgres.rs

```diff
@@ -250,6 +252,35 @@ pub fn rows_to_arrow(rows: &[Row]) -> Result<RecordBatch> {
                 Type::BOOL => {
                     handle_primitive_type!(builder, Type::BOOL, BooleanBuilder, bool, row, i);
                 }
+                Type::TIME => {
+                    let Some(builder) = builder else {
+                        return NoBuilderForIndexSnafu { index: i }.fail();
+                    };
+                    let Some(builder) = builder
+                        .as_any_mut()
+                        .downcast_mut::<Time64NanosecondBuilder>()
+                    else {
+                        return FailedToDowncastBuilderSnafu {
+                            postgres_type: format!("{postgres_type}"),
+                        }
+                        .fail();
+                    };
+                    let v = row
+                        .try_get::<usize, Option<chrono::NaiveTime>>(i)
+                        .with_context(|_| FailedToGetRowValueSnafu {
+                            pg_type: Type::TIME,
+                        })?;
+
+                    match v {
+                        Some(v) => {
```

Comment:
Let's add a test for this logic

Reply:
Test added for time chrono -> arrow time64nanosecond logic